### DISCUSSION TITLE
Issue with docks?

### DISCUSSION THREAD

**Discussion by catalin-duta** [Keriew/augustus#652] [Q&A]
I'm playing on the Tarsus map and I noticed that for some 2 years trading boats have become rare and few in between.
Sometimes I see a boat spawning (NW side) and it disappears right away.
It seems to improve (not sure if 100% spawns) if I demolish the first dock (awkwardly placed, yes).
Has anyone else noticed something similar?
Screenshot and save attached.
![Screenshot_20220820-211417328](https://user-images.githubusercontent.com/93785920/185761001-660de089-643e-419b-8705-6ac8de4a04c4.jpg)
[7TarsusA2j.zip](https://github.com/Keriew/augustus/files/9387811/7TarsusA2j.zip)

**Comment by crudelios** [comment]
If you're playing on 3.1.0, this issue has likely been fixed on the latest unstable!

**Reply by catalin-duta** [reply]
I'm on 3.1.0.514, what looks like 3 commits back - and those seem unrelated.

**Comment by Ouaz** [comment]
I tested the save on last unstable and I confirm the bug:

![pop_ship_plop](https://user-images.githubusercontent.com/11946570/185763539-ba6e812a-c20d-4e50-b049-c0dfc9f174c9.jpg)

EDIT: if the left dock is set "to not accept" ships from Miletus or Alexandria, the trade ships spawn correctly and go to the dock on the right. It's the same than destroying the dock it actually. ^^

**Comment by crudelios** [comment]
Sigh, yet another dock issue.. this happens because the dock is oriented towards a canal, so the ship doesn't find a proper queuing spot, preventing it from spawning.

The only solution for now is to remove that dock. I'll eventually come up with a more dynamic way to find valid ship queuing spots, but not before v4.0.

**Edit:** created an issue for this discussion. Please check #653.
